### PR TITLE
Allow Vault Agent to run as a Windows service

### PR DESCRIPTION
**Pull request by raskchanky**
Closes https://github.com/hashicorp/vault/issues/7609

### CODE REVIEW

File: command/agent/winsvc/service_windows.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+package winsvc
+
+import (
+	wsvc "golang.org/x/sys/windows/svc"
+)
+
+type serviceWindows struct{}
+
+func init() {
+	interactive, err := wsvc.IsAnInteractiveSession()
```
Comment (calvn) [inline command/agent/winsvc/service_windows.go:12]: Looks like this func [is deprecated](https://godoc.org/golang.org/x/sys/windows/svc#IsAnInteractiveSession), and the recommendation is to use `IsWindowsService` instead. Found an example of the usage in [here](https://github.com/golang/sys/blob/master/windows/svc/example/main.go#L39).

**Review by calvn — APPROVED** [review]
TIL. This is pretty cool, thanks for working on this!

One comment/suggestion but otherwise looks good.

File: command/agent/winsvc/service_windows.go (hunk 1 above)
Comment (raskchanky) [reply command/agent/winsvc/service_windows.go:12]: Ah, good catch, thanks!

**Comment by calvn** [comment]
@raskchanky Looks like CI is not happy that the package is not vendored.

```
command/agent/winsvc/service_windows.go:6:2: cannot find package "golang.org/x/sys/windows/svc" in any of:
	/gopath/src/github.com/hashicorp/vault/vendor/golang.org/x/sys/windows/svc (vendor tree)
	/goroot/src/golang.org/x/sys/windows/svc (from $GOROOT)
	/gopath/src/golang.org/x/sys/windows/svc (from $GOPATH)
```

File: command/agent/winsvc/service_windows.go (hunk 1 above)
Comment (raskchanky) [reply command/agent/winsvc/service_windows.go]: It turns out `IsWindowsService` hasn't been released yet. I updated this PR to use it and it won't compile, even with go 1.15.4. I'm going to revert that change.

File: CHANGELOG.md [hunk 2]
```diff
@@ -955,7 +960,7 @@ KNOWN ISSUES:
 * OSS binaries of 1.5.1, 1.4.4, 1.3.8, and 1.2.5 were built without the Vault UI. Enterprise binaries are not affected.
 * AWS IAM logins may return an error depending on the headers sent with the request.
   For more details and a workaround, see the [1.2.5 Upgrade Guide](https://www.vaultproject.io/docs/upgrading/upgrade-to-1.2.5)
-  
+
 BUG FIXES:
 * seal: (enterprise) Fix issue causing stored seal and recovery keys to be mistaken as sealwrapped values
```
Comment (raskchanky) [inline CHANGELOG.md:966]: Apologies for the spam in this file. My editor is configured to strip all trailing whitespace on save.

**Pull request merged**